### ISSUE TITLE
Compile errors

### ISSUE THREAD

**Issue by realjax** [pcloudcom/console-client#71]
On Ubuntu 18.04

when running 'make fs'
`pdevice_monitor.c:338:10: fatal error: libudev.h: No such file or directory`

when running last 'cmake .'
`/usr/include/boost/smart_ptr/detail/local_counted_base.hpp:67:9: error: ‘__builtin_assume’ was not declared in this scope
         __builtin_assume( local_use_count_ >= 1 );
         ^~~~~~~~~~~~~~~~
/usr/include/boost/smart_ptr/detail/local_counted_base.hpp:67:9: note: suggested alternative: ‘__builtin_abs’
         __builtin_assume( local_use_count_ >= 1 );
         ^~~~~~~~~~~~~~~~
         __builtin_abs
CMakeFiles/pcloudcc_lib.dir/build.make:86: recipe for target 'CMakeFiles/pcloudcc_lib.dir/pclsync_lib.cpp.o' failed
make[2]: *** [CMakeFiles/pcloudcc_lib.dir/pclsync_lib.cpp.o] Error 1
CMakeFiles/Makefile2:168: recipe for target 'CMakeFiles/pcloudcc_lib.dir/all' failed
make[1]: *** [CMakeFiles/pcloudcc_lib.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2`

when running 'sudo make install' the error  above was repeated

then 'ldconfig' needed root privilges.

and finally (and obviously :) ) 'pcloudcc ' could not be found.

**Comment by realjax** [comment]
Still stuck on the second errror: __builtin_assume’ was not declared in this scope

**Comment by plubinski** [comment]
Any solution?

**Comment by realjax** [comment]
Yes I got it to compile in the end. But I can't remember what it was that I did... I think some packages failed to install and reinstalling did the trick or something.

**Comment by jaromrax** [comment]
I am stuck with this. Ubuntu 18.04.1, fresh install, all prerequisites, also libudev-dev.

**Comment by hevanaa** [comment]
I'm trying to build on Fedora 28 and get similar errors. This is when running make in the root directory. The first error is:


```
In file included from /usr/include/c++/8/bits/unique_ptr.h:36,
                 from /usr/include/c++/8/memory:80,
                 from /usr/include/boost/config/no_tr1/memory.hpp:21,
                 from /usr/include/boost/smart_ptr/shared_ptr.hpp:23,
                 from /usr/include/boost/shared_ptr.hpp:17,
                 from /home/<undisclosed path>/console-client/pCloudCC/pclsync_lib.cpp:45:
/usr/include/c++/8/utility:305:22: error: ‘__make_integer_seq’ does not name a type; did you mean ‘__make_unsigned’?
       using __type = __make_integer_seq<_IdxTuple, size_t, _Num>;
                      ^~~~~~~~~~~~~~~~~~
                      __make_unsigned
/usr/include/c++/8/utility:327:9: error: ‘__make_integer_seq’ does not name a type; did you mean ‘__make_unsigned’?
       = __make_integer_seq<integer_sequence, _Tp, _Num>;
         ^~~~~~~~~~~~~~~~~~
         __make_unsigned
/usr/include/c++/8/utility:340:33: error: ‘make_integer_sequence’ does not name a type; did you mean ‘integer_sequence’?
     using make_index_sequence = make_integer_sequence<size_t, _Num>;
                                 ^~~~~~~~~~~~~~~~~~~~~
                                 integer_sequence
/usr/include/c++/8/utility:344:32: error: ‘make_index_sequence’ does not name a type; did you mean ‘index_sequence’?
     using index_sequence_for = make_index_sequence<sizeof...(_Types)>;
                                ^~~~~~~~~~~~~~~~~~~
                                index_sequence
```

Later on there is the "__builtin_assume was not declared in this scope" mentioned above.

**Comment by hevanaa** [comment]
Is maybe "error: ‘__make_integer_seq’" a compiler issue? Is it only available in clang?

**Comment by hevanaa** [comment]
To make cmake work I had to set Boost_USE_STATIC_LIBS OFF. 

Using the source at https://github.com/plague-doctor/console-client I am able to compile the client.

**Comment by mejiasjrg** [comment]
First error I got was the "libudev.h: No such file or directory", which was solved by installing libudev-dev.
Now I'm getting this:
cc -Wall -Wpointer-arith -O2 -g -fno-stack-protector -fomit-frame-pointer -fPIC -DP_OS_LINUX -I../sqlite -DP_SSL_MBEDTLS -I../mbedtls/include   -c -o pssl-mbedtls.o pssl-mbedtls.c
pssl-mbedtls.c: In function ‘psync_ssl_init’:
pssl-mbedtls.c:102:3: error: PIC register clobbered by ‘%ebx’ in ‘asm’
   __asm__("cpuid"
   ^
make: *** [pssl-mbedtls.o] Error 1
What should I do?
Thanks.
